File: mediremind_mobile/lib/services/auth_service.dart
import '../data/repositories/auth_repository.dart';
import '../models/app_user.dart';

class AuthService {
  AuthService({AuthRepository? repository})
    : _repository = repository ?? AuthRepository();

  final AuthRepository _repository;

  Future<void> register({
    required String email,
    required String password,
    required String fullName,
  }) {
    return _repository.register(
      email: email.trim(),
      password: password,
      fullName: fullName.trim(),
    );
  }

  Future<AppUser> login({
    required String email,
    required String password,
  }) {
    return _repository.login(email: email.trim(), password: password);
  }

  Future<void> sendPasswordResetEmail(String email) {
    return _repository.sendPasswordResetEmail(email.trim());
  }

  Future<void> logout() {
    return _repository.logout();
  }
}

File: backend/src/services/authService.ts
import bcrypt from "bcryptjs";
import { User } from "../models/user";
import { UserRepository } from "../repositories/userRepository";

export class AuthService {
  constructor(private readonly repository = new UserRepository()) {}

  async register(
    email: string,
    password: string,
    fullName: string,
  ): Promise<User> {
    const existing = await this.repository.findByEmail(email);
    if (existing) {
      throw new Error("Email đã được đăng ký.");
    }

    const hashedPassword = await bcrypt.hash(password, 10);
    return this.repository.create({ email, fullName }, hashedPassword);
  }
}
